python-package/lightgbm/plotting.py

```diff
@@ -268,7 +268,8 @@ def _to_graphviz(tree_info, show_info, feature_names, precision=None,
         raise ImportError('You must install graphviz to plot tree.')
 
     def float2str(value, precision=None):
-        return "{0:.{1}f}".format(value, precision) if precision is not None else str(value)
+        return "{0:.{1}f}".format(value, precision) \
+            if (precision is not None) and (not isinstance(value, str)) else str(value)
```

Comment:
Please remove the brackets and use `string_type` from `compat.py` module instead of `str` in `isinstance()`.